media/Jsontracer/index.js

```diff
@@ -0,0 +1,160 @@
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This file referenced the result of https://github.com/catapult-project/catapult/tree/444aba89e1c30edf348c611a9df79e2376178ba8/tracing
+
+// TODO import openFileSelector from "./processData.js";
+// TODO import dynamicGraduation from "./dynamicGraduation.js";
+
+const graph = document.querySelector(".graph");
+const sliderMaxLimit = 6400;
+const sliderMinLimit = 100;
+
+const loadBtn = document.querySelector(".load-btn");
+loadBtn.addEventListener("click", () => {
+  initData();
+  // TODO openFileSelector();
+});
+
+let ratio = 100;
+const slider = document.querySelector("input");
+slider.addEventListener("input", event => {
+  ratio = event.target.value * 1;
+  graph.style.width = `${ratio}%`;
+  changeSlider(event.target.value, event.target.max, event.target.min);
+  // TODO dynamicGraduation();
+});
+
+const zoomInBtn = document.querySelector(".zoom-in-btn");
+zoomInBtn.addEventListener("click", () => {
+  if (ratio >= 5000) { return; }
+  
+  // change ratio
+  ratio += 50;
+  ratio = ratio > 5000 ? 5000 : ratio;
+  graph.style.width = `${ratio}%`;
+
+  // change slider
+  const slider = document.querySelector("input");
+  slider.value = ratio;
+  changeSlider(slider.value, slider.max, slider.min);
+
+  // change graduation and set delay
+  dynamicGraduation();
+  zoomInBtn.disabled = true;
+  setTimeout(() => (zoomInBtn.disabled = false), 300);
+});
+
+const zoomOutBtn = document.querySelector(".zoom-out-btn");
+zoomOutBtn.addEventListener("click", () => {
+  if (ratio <= 100) { return; }
+
+  // change ratio
+  ratio -= 50;
+  ratio = ratio < 100 ? 100 : ratio;
+  graph.style.width = `${ratio}%`;
+
+  // change slider
+  const slider = document.querySelector("input");
+  slider.value = ratio;
+  changeSlider(slider.value, slider.max, slider.min);
+
+  // change graduation and set delay
+  dynamicGraduation();
+  zoomOutBtn.disabled = true;
+  setTimeout(() => (zoomOutBtn.disabled = false), 300);
+});
+
+const captureBtn = document.querySelector(".capture-btn");
+captureBtn.addEventListener("click", () => {
+  // TODO capture inside vscode extension webview
+});
+
+function changeSlider(inputValue, inputMax, inputMin) {
+  if (inputMax === inputValue) {
+    slider.max = String(Math.min(inputMax*2, sliderMaxLimit));
+    slider.value = String(inputMax);
+    slider.min = String(inputMax*0.5);
+  } else if (inputMin === inputValue) {
+    slider.max = String(inputMin*2);
+    slider.value = String(inputMin);
+    slider.min = String(Math.max(inputMin*0.5, sliderMinLimit));
+  } else {
+    return;
+  }
+
+  // set delay
+  slider.disabled = true;
+  setTimeout((slider.disabled = false), 100);
+}
+
+function initData() {
+  const graph = document.querySelector(".graph");
```

Comment:
Doesn't this conflict with `const graph = document.querySelector(".graph");` in line 50?

Reply:
It doesn't collide because it's a function scope, but it's a duplicate code, so I'll delete it. Thank you.